test/extended/etcd/helpers/helpers.go

```diff
@@ -43,41 +43,50 @@ type TestingT interface {
 	Logf(format string, args ...interface{})
 }
 
-// CreateNewMasterMachine creates a new master node by cloning an existing Machine resource.
-// machineIndex is used to identify the index to which machine should be created to.
-// machineIndex of -1 to be passed if index needn't be taken into consideration while creating the machine
-func CreateNewMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface, machineIndex int64) (string, error) {
+// GetRunningMasterMachine finds and returns a running master machine from the cluster.
+func GetRunningMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (machinev1beta1.Machine, error) {
```

Comment:
```suggestion
func AnyRunningMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (machinev1beta1.Machine, error) {
```

maybe I've been doing too much C#, but this just gets any running Machine, so it makes a bit more sense :)